File: Projects/rosalind/DNA.java
import java.io.*;
import java.util.*;

public class DNA {
	private static final char[] DNABASES = new char[] { 'A', 'C', 'G', 'T' };
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(System.in);
		String seq = in.nextLine();
		for (final char base : DNABASES)
			System.out.println(seq.chars().filter(x -> x == base).count());
		in.close();
	}
}

File: Projects/rosalind/DNA.kt
package rosalind

import java.io.IOException
import java.util.*

object DNA {
    private val DNABASES = charArrayOf('A', 'C', 'G', 'T')
    @Throws(IOException::class)
    @JvmStatic
    fun main(args: Array<String>) {
        val stdin = Scanner(System.`in`)
        val seq = stdin.nextLine()
        for (base in DNABASES) println(seq.chars().filter { x: Int -> x == base.toInt() }.count())
        stdin.close()
    }
}